src/taco-cli/cli/kit.ts

```diff
@@ -452,12 +456,18 @@ class Kit extends commands.TacoCommandBase {
     /**
      * Updates the platforms and plugins added to the project - after a kit/cli change
      */
-    private static updateProject(editParams: ITacoJsonEditParams, installedPlatformVersions: IDictionary<string>, installedPluginVersions: IDictionary<string>,
-        platformVersionUpdates: IDictionary<string> = null, pluginVersionUpdates: IDictionary<string> = null): Q.Promise<any> {
+    private static updateProject(editParams: ITacoJsonEditParams, installedPlatformVersions: IDictionary<string>,
+        installedPluginVersions: IDictionary<string>, platformVersionUpdates: IDictionary<string> = null,
+        pluginVersionUpdates: IDictionary<string> = null, savedPluginVersionUpdates: IDictionary<string> = null): Q.Promise<any> {
         logger.log(resources.getString("CommandKitSelectStatusUpdatingTacoJson"));
         // First, update the taco.json to reflect the new CLI/kit version
         // Then go ahead and update the platforms and plugins
         return kitHelper.editTacoJsonFile(editParams)
+        .then(() => ProjectHelper.getProjectInfo())
+        .then(projectInfo => {
+            var plugins: Cordova.ICordovaPluginInfo[] = Object.keys(savedPluginVersionUpdates).map(pluginName => { return { name: pluginName, spec: null }; });
+            return CordovaHelper.editPluginVersionSpecs(plugins, projectInfo);
```

Comment:
I added:
            // We remove the plugins that are overriden by the target kit from the config.xml
            // because the config.xml has preference over the kit, and if we don't do this we'll
            // break the kit. We need also to do this before changing the platforms, because changing
            // a platforms automatically adds plugins found on the config.xml file